test/helpers/erc4337.js

```diff
@@ -80,6 +88,42 @@ class SmartAccount extends ethers.BaseContract {
   }
 }
 
+class ERC7702SmartAccount extends ethers.BaseContract {
+  constructor(instance, signer, context) {
+    super(signer.address, instance.interface, instance.runner, undefined);
+    this.address = signer.address;
+    this.delegate = instance;
+    this.context = context;
+  }
+
+  async deploy() {
+    await ethers.provider.getCode(this.delegate).then(code => setCode(this.address, code));
+    return this;
+  }
+
+  async createOp(args = {}) {
+    await this.context.wait();
+
+    const params = Object.assign({ sender: this }, args);
+    // fetch nonce
+    if (!params.nonce) {
+      params.nonce = await this.context.entrypoint.getNonce(this, 0);
+    }
+    // prepare paymaster and data
+    if (ethers.isAddressable(params.paymaster)) {
+      params.paymaster = await ethers.resolveAddress(params.paymaster);
+      params.paymasterVerificationGasLimit ??= 100_000n;
+      params.paymasterPostOpGasLimit ??= 100_000n;
+      params.paymasterAndData = ethers.solidityPacked(
+        ['address', 'uint128', 'uint128'],
+        [params.paymaster, params.paymasterVerificationGasLimit, params.paymasterPostOpGasLimit],
+      );
+    }
+
+    return new UserOperationWithContext(params);
+  }
```

Comment:
I don't like this logic is duplicated with `SmartAccount` but I see it'd be too much to abstract away only this function.

Works for me

Reply:
Added a fillUserOp in the helper. Let me know what you think

Reply:
I like it, thanks!